client/state/sites/hooks/use-site-admin-interface-data.ts

```diff
@@ -14,7 +14,9 @@ const useSiteAdminInterfaceData = ( siteId: number = 0 ) => {
 	const wpcomAdminInterface = useSelector( ( state: AppState ) =>
 		getSiteOption( state, siteId, 'wpcom_admin_interface' )
 	);
-	const isJetpack = useSelector( ( state: AppState ) => isJetpackSite( state, siteId ) );
+	const isJetpack = useSelector( ( state: AppState ) =>
+		isJetpackSite( state, siteId, { treatAtomicAsJetpackSite: false } )
+	);
 	const isWPAdmin = wpcomAdminInterface === 'wp-admin' || isJetpack;
```

Comment:
Previously, `isJetpack` would be true for atomic sites, this is why they were showing links as if they had the 'wp-admin' option regardless of their actual setting.